### PR TITLE
update geojson, contours. Use new FeatureWriter API

### PR DESCRIPTION
**Pull request by michaelkirk**
The new [geojson::FeatureWriter](https://docs.rs/geojson/latest/geojson/struct.FeatureWriter.html) API allows sharing some code between the CSV and GeoJSON outputs.

Let me know what you think!

### CODE REVIEW

File: apps/ltn/src/impact/ui.rs [hunk 1]
```diff
@@ -422,26 +425,7 @@ fn export_csv(app: &App) -> Result<String> {
...
-                    .get(r.id),
-                total_count_after: app
-                    .per_map
-                    .impact
-                    .compare_counts
-                    .counts_b
-                    .per_road
-                    .get(r.id),
-            })?;
+            writer.serialize(ExportRow::new(r, app))?
```
Comment (dabreegster) [inline apps/ltn/src/impact/ui.rs:428]:
```suggestion
            writer.serialize(ExportRow::new(r, app))?;
```
The semicolon missing weird me out a bit O_O

File: apps/ltn/src/impact/ui.rs [hunk 2]
```diff
@@ -459,39 +443,60 @@ struct ExportRow {
...
+        for r in app.per_map.map.all_roads() {
+            let bounds = app.per_map.map.get_gps_bounds();
+            let mut geometry = geo::LineString::from(&r.center_pts);
+            geometry.map_coords_in_place(|c| {
+                let lonlat = bounds.convert_back_xy(c.x, c.y);
+                return geo::coord! { x: lonlat.x(), y: lonlat.y() };
+            });
+
+            let sr = RoadGeoJson {
+                export_row: ExportRow::new(r, app),
```
Comment (dabreegster) [inline apps/ltn/src/impact/ui.rs:493]: Sharing the properties here is indeed great, nice API!

File: geom/src/bounds.rs [hunk 3]
```diff
@@ -231,6 +231,17 @@ impl GPSBounds {
         pts.iter().map(|pt| pt.to_gps(self)).collect()
     }
 
+    pub fn convert_back_xy(&self, x: f64, y: f64) -> LonLat {
+        let (width, height) = {
```
Comment (dabreegster) [inline geom/src/bounds.rs:235]: What if we just did `Pt2D::new(x, y).to_gps(self)` here? Any reason to copy the math?

Comment (dabreegster) [reply geom/src/bounds.rs:235]: Edit: you moved the logic here, this is fine!

**Review by dabreegster — APPROVED** [review]
Thank you! The new writer API is great

File: Cargo.lock [hunk 4]
```diff
@@ -1594,7 +1608,7 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "a8850d78e5f2f803e3b201805bf41aea28eeff71bd0b56c3ff067e805d208b58"
 dependencies = [
  "geo-types",
- "geojson",
+ "geojson 0.23.0",
```
[the remark this answers is not in the document]
Comment (michaelkirk) [reply Cargo.lock]: Whoops good catch! Fixed in ebf0f994f0

**Review by dabreegster — APPROVED** [review]
Thanks for the PR!

**Pull request merged**